tests/testthat/test_postgis_RPostgres.R

```diff
@@ -0,0 +1,320 @@
+    expect_silent(x <- st_read("PG:host=localhost dbname=postgis", "meuse_multi", quiet = TRUE, type = c(4,4)))
+    expect_silent(x <- st_read("PG:host=localhost dbname=postgis", "meuse_multi", quiet = TRUE, promote_to_multi = FALSE))
+    expect_silent(x <- st_read("PG:host=localhost dbname=postgis", "meuse_multi", quiet = TRUE, geometry_column = "geometry.1"))
+    x <- st_layers("PG:host=localhost dbname=postgis")
+    multi2 <- cbind(pts[["geometry"]], st_set_crs(st_transform(pts, 4326), NA))
+    expect_silent(st_write(multi2, pg, "meuse_multi2", overwrite = TRUE))
+    expect_silent(x <- st_read(pg, "meuse_multi2"))
+    expect_equal(st_crs(x[["geometry"]]), st_crs(multi2[["geometry"]]))
+    expect_equal(st_crs(x[["geometry.1"]]), st_crs(multi2[["geometry.1"]]))
+    expect_silent(x <- st_read("PG:host=localhost dbname=postgis", "meuse_multi2", quiet = TRUE))
+    #expect_equal(st_crs(x[["geometry"]]), st_crs(multi2[["geometry"]]))
+    expect_equal(st_crs(x[["geometry.1"]]), st_crs(multi2[["geometry.1"]]))
+})
+
+test_that("sf can write units to database (#264)", {
+    skip_if_not(can_con(pg), "could not connect to postgis database")
+    ptsu <- pts
+    ptsu[["u"]] <- ptsu[["cadmium"]]
+    units(ptsu[["u"]]) <- units::make_unit("km")
+    expect_silent(st_write(ptsu, pg, "sf_units__", overwrite = TRUE))
+    r <- st_read(pg, "sf_units__")
+    expect_is(r[["u"]], "numeric")
+    expect_equal(sort(r[["u"]]), sort(as.numeric(ptsu[["u"]])))
+    dbRemoveTable(pg, "sf_units__")
+})
+
+test_that("sf can preserve types (#592)", {
+    skip_if_not(can_con(pg), "could not connect to postgis database")
+    dtypes <- data.frame(
+        logi = c(TRUE, FALSE, NA),
+        chara = c("a", "", NA),
+        nume = c(1.1e1, 2.2e2, NA),
+        inte = c(1L, 2L, NA),
+        fact = factor(c("a", "b", NA), levels = letters),
+        #comp = c(complex(1, 2), complex(2, 3)),
+        date = c(rep(Sys.Date(), 2), NA),
+        time = c(rep(Sys.time(), 2), NA),
+        x = c(1, 2, 4),
+        y = c(1, 2, 4), stringsAsFactors = FALSE)
+    # cannot write lists
+    #dtypes$lst <- c(list(matrix("a")), list(matrix(c("b", "c"))), list(NA))
+    dtypes <- st_as_sf(dtypes, coords = c("x", "y"))
+    st_write(dtypes, pg, overwrite = TRUE)
+    x <- st_read(pg, "dtypes")
+    dtypes$fact <- as.character(dtypes$fact)
+    expect_equal(x, dtypes)
+    DBI::dbRemoveTable(pg, "dtypes")
+})
+
+test_that("can write to other schema", {
+    skip_if_not(can_con(pg), "could not connect to postgis database")
+    try(DBI::dbSendQuery(pg, "CREATE SCHEMA sf_test__;"), silent = TRUE)
+    q <- "SELECT schema_name FROM information_schema.schemata WHERE schema_name = 'sf_test__';"
+    suppressWarnings(could_schema <- DBI::dbGetQuery(pg, q) %>% nrow() > 0)
+
+    skip_if_not(could_schema, "Could not create schema (might need to run 'GRANT CREATE ON DATABASE postgis TO <user>')")
+    expect_error(st_write(pts, pg, Id(schema = "public", table = "sf_meuse__")), "exists")
```

Comment:
I'm getting
```r
Error: `st_write(pts, pg, Id(schema = "public", table = "sf_meuse__"))` threw an error with unexpected message.
Expected match: "exists"
Actual message: "unable to find an inherited method for function ‘dbWriteTable’ for signature ‘\"PqConnection\", \"SQL\", \"missing\"’"
```
here -- any idea?

Reply:
Make sure you are running the latest `DBI` and `RPostgres`.

Reply:
I did, but that didn't resolve it; sessionInfo below; postgresql 9.6.8. 

When I outcomment these test, next thing I get stuck with is 
```r
> st_write(pts, pg, Id(schema = "CAP__", table = "Meuse_tbl"))
Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function ‘dbWriteTable’ for signature ‘"PqConnection", "SQL", "missing"’
```
What am I missing?
<details>

```r
> sessionInfo()
R version 3.4.3 (2017-11-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 17.10

Matrix products: default
BLAS: /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] sp_1.2-7       testthat_2.0.0 DBI_0.8       

loaded via a namespace (and not attached):
 [1] bit_1.1-12      compiler_3.4.3  magrittr_1.5    R6_2.2.2       
 [5] hms_0.4.2       tools_3.4.3     Rcpp_0.12.16    bit64_0.9-7    
 [9] grid_3.4.3      blob_1.1.1      pkgconfig_2.0.1 rlang_0.2.0    
[13] RPostgres_1.1.0 lattice_0.20-35
```
</details>

Reply:
This error is because the test uses the new way of naming tables in `DBI`. It uses `DBI::Id()` (prior was `DBI::SQL()`). Also `Rpostgres` was just recently updated (https://github.com/r-dbi/RPostgres/issues/179) could you try `devtools::install_github("r-dbi/rpostgres")`. I was surprised that travis ran the test without complaining.

Looking at the travis logs (`devtools::session_info()`)
```
DBI              0.8       2018-03-02 cran (@0.8)
```
Can't find `RPostgres`, but it must be 1.1.0. (I'm using `RPostgres 1.1.0 2018-04-05 from Github (r-dbi/RPostgres@ef2ab88)`)

Reply:
This is probably not causing this, but in general, shouldn't `RPostgres` be in `Suggests` in `DESCRIPTION`, (preferably with minimal version number)? Or am I missing something?

Reply:
Good question. I don't know. I think the whole point of using DBI is to only require DBI, but it's true that we sould still suggest RPostgres, but then should we suggest other drivers?

Reply:
If there are tests for them, they ought to be added, right?

Reply:
For our own education, I searched a bit for other examples and I found that `dplyr` uses [`data.table` in tests](https://github.com/tidyverse/dplyr/blob/892cf876ec8044f1ec6f5ca7904f939fd9281262/tests/testthat/test-group-by.r#L134), but it's nowhere in the [`DESCRIPTION`](https://github.com/tidyverse/dplyr/blob/master/DESCRIPTION). I still think it's a good idea to suggest it, though. Maybe @krlmlr knows the reason?

Reply:
We moved the data.table backend to *dtplyr* a while ago. The tests are gone now: https://github.com/tidyverse/dplyr/blob/master/tests/testthat/test-group-by.r.

Reply:
Ah, thanks (I got confused in the commits)! Makes sense. So if I understand correctly we should `Import` `DBI` and `Suggest` all drivers we explicitly test.